src/pre_process/m_global_parameters.fpp

```diff
@@ -779,12 +797,32 @@ contains
         chemxb = species_idx%beg
         chemxe = species_idx%end
 
+        ! Determining the number of cells that are needed in order to store
+        ! sufficient boundary conditions data as to iterate the solution in
+        ! the physical computational domain from one time-step iteration to
+        ! the next one
+        if (viscous) then
+            buff_size = 2*weno_polyn + 2
```

Comment:
this is duplicate code from simulation. move it somewhere common, along with all duplicate code